libbeat/outputs/elasticsearch/output.go

```diff
@@ -117,6 +120,30 @@ func (out *elasticsearchOutput) init(
 		return err
 	}
 
+	// Check if template should be loaded
+	// Not being able to load the template will output an error but will not stop execution
+	if config.Template.Name != "" && len(clients) > 0 {
+
+		esClient := clients[0].(*Client)
+
+		logp.Debug("test", "Test: %v", config.Template)
+
+		// Check if template already exist or should be overwritten
+		if !esClient.CheckTemplate(config.Template.Name) && config.Template.Overwrite {
```

Comment:
There is a race condition with `CheckTemplate` and `LoadTemplate` when using multiple Beats (potentially of different versions) but I don't know of any way to work-around it.